Deprecate synced flush (#50835)
elastic/elasticsearch: docs/reference/upgrade/synced-flush.asciidoc

--------------------------------------------------
POST _flush/synced
--------------------------------------------------
// TEST[skip: will fail as synced flush is deprecated]

When you perform a synced flush, check the response to make sure there are
no failures. Synced flush operations that fail due to pending indexing
operations are listed in the response body, although the request itself
still returns a 200 OK status. If there are failures, reissue the request.

> Note that synced flush is deprecated and will be removed in future versions.

Reviewer: ```suggestion
NOTE: Synced flush is deprecated and will be removed in a future version.
```

> A normal flush has the same effect as a synced flush on Elasticsearch 7.6

Reviewer: ```suggestion
Flush has the same effect as synced flush in {es} 7.6
```

> or later.

Reviewer: ```suggestion
and later versions.
```
